### ISSUE TITLE
Pyright should report `@runtime_checkable` on non-protocol as error

### ISSUE THREAD

**Issue by A5rocks** [microsoft/pyright#11715]
**Describe the bug**
See title.

**Code or Screenshots**

```python
from typing import runtime_checkable

@runtime_checkable
class Bweh: ...
```

**VS Code extension or command-line**
Pyright playground

**Comment by bschnurr** [comment]
Thanks for the report. This is covered by #11131, and #11646 contains a fix to reject @runtime_checkable when it is applied to a non-protocol class.

I’m going to track the work there and close this as a duplicate.
